code/modules/clothing/spacesuits/soviet.dm

```diff
@@ -74,22 +77,39 @@
 
 /datum/component/armor_protection_limb/proc/on_equip(obj/item/I, mob/equipper)
 	RegisterSignal(equipper, COMSIG_ATOM_BULLET_ACT, .proc/on_projectile)
+	if(has_overlay)
+		update_overlay()
 
 /datum/component/armor_protection_limb/proc/on_unequip(obj/item/I, mob/unequipper, slot)
 	if(isitem(parent))
 		if(!(I.flags_equip_slot & slotdefine2slotbit(slot)))
 			return
+		if(has_overlay)
+			usr.clear_fullscreen("helmet")
 	UnregisterSignal(unequipper, COMSIG_ATOM_BULLET_ACT)
 
 /datum/component/armor_protection_limb/proc/on_projectile(obj/item/I, var/obj/projectile/proj)
 	if(covered_limbs & GLOB.string_part_flags[proj.def_zone])
 		return adjust_health(max(0, proj.damage - round(proj.distance_travelled * proj.damage_falloff)))
 	return FALSE
 
+/datum/component/armor_protection_limb/proc/update_overlay(mob/user)
+	if(health <= 90) //this is taped together but what in this game isn't
+		usr.overlay_fullscreen("helmet", /obj/screen/fullscreen/helmet, 1)
+	if(health <= 60)
+		usr.overlay_fullscreen("helmet", /obj/screen/fullscreen/helmet, 2)
+	if(health <= 30)
+		usr.overlay_fullscreen("helmet", /obj/screen/fullscreen/helmet, 3)
+	if(health <= 5)
+		usr.overlay_fullscreen("helmet", /obj/screen/fullscreen/helmet, 4)
+
```

Comment:
This section of if statements is 'bad' and would be improved by being a `switch` statement instead.

Here is an example

```
					switch(M.reagent_pain_modifier)
						if(PAIN_REDUCTION_MEDIUM to PAIN_REDUCTION_HEAVY)
							multipler += 0.15
						if(PAIN_REDUCTION_HEAVY to PAIN_REDUCTION_VERY_HEAVY)
							multipler += 0.25
						if(PAIN_REDUCTION_VERY_HEAVY to PAIN_REDUCTION_FULL)
							multipler += 0.40
						if(PAIN_REDUCTION_FULL to INFINITY)
							multipler += 0.45
```